# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/RenderRootTypes.h
# Commit: f6e400fb14a2
# Full Hash: f6e400fb14a2bd8133ba6a83d1b1b233fc41bf99
# Author: Alexis Beingessner <a.beingessner@gmail.com>
# Date: 2019-08-21 21:53:43
# Description:
#   Bug 1569131 - Also defer RecvParentCommands. r=jrmuizel
#   
#   We defer ParentCommands when they come in attached to a display list,
#   but we forgot to defer them when they're sent directly. This was
#   causing us to crash when we defer updates that register textures
# ==============================================================================

diff -r 8c915889c8f0 -r f6e400fb14a2 gfx/layers/wr/RenderRootTypes.h
--- a/gfx/layers/wr/RenderRootTypes.h	Wed Aug 21 07:38:13 2019 +0000
+++ b/gfx/layers/wr/RenderRootTypes.h	Wed Aug 21 13:59:49 2019 +0000
@@ -48,8 +48,12 @@
   nsTArray<mozilla::ipc::Shmem> mLargeShmems;
 };
 
+struct ParentCommands {
+  nsTArray<WebRenderParentCommand> mCommands;
+};
+
 typedef Variant<RenderRootDisplayListData, RenderRootUpdates, ResourceUpdates,
-                FocusTarget>
+                ParentCommands, FocusTarget>
     RenderRootDeferredData;
 
 }  // namespace layers